File: algorithm/405. Convert a Number to Hexadecimal/405.cpp
class Solution {
public:
    string toHex(int num) {
        if (num == 0) return "0";
        string result="";
        int count=0;
        while(num && count<8)
        {
            int tem=num & 15;
            char a;
            if(tem<10)
            {
                a=('0'+tem);
                result=a+result;
            }
            else
            {
                a=('a'+tem-10);
                result= a+result;
            }
            num = num>>4;
            count++;
        }
        return result;
    }
};

File: algorithm/405. Convert a Number to Hexadecimal/405.py
class Solution(object):
    def toHex(self, num):
        """
        :type num: int
        :rtype: str
        """
        result=""
        dic = {10:"a", 11:"b", 12:"c", 13:"d", 14:"e", 15:"f"}
        if num==0:
            return '0'
        if num < 0:
            num=num+2**32
        while num>0:
            if(num%16<10):
                result= str(num%16)+result
            else:
                result = dic[num%16] +result
            num=num/16
        return result
